### ISSUE TITLE
dockerfile inheritance base/devel

### ISSUE THREAD

**Issue by bebehei** [nubs/docker-arch-build#1]
Hi,

what about basing this image off from `finalduty/archlinux`? This has the improvement to make Dockerfile simpler and additional it shrinks the overall image size, as `finalduty/archlinux` is regularly updated in comparison to `base/archlinux`.

Just the difference of a self-built image based on finalduty's image:
```
REPOSITORY            TAG                 IMAGE ID            CREATED             SIZE
archlinux             latest              3d85b69c6b98        8 minutes ago       560.3 MB
finalduty/archlinux   latest              028d86c90268        6 hours ago         333.3 MB
nubs/arch-build       latest              a68cd3855bfe        3 weeks ago         1.282 GB
```

Also, I might have to clearify: Because of the absence of an official docker image, most people get tricked into thinking `base/` might be official. It pops always up as first result, but in fact, it's just a normal (sneaky) username.

**Comment by nubs** [comment]
Thanks for the suggestion!  I'd be willing to investigate it further but I'd rather wait for an actual official archlinux image to use.  You are correct that there is nothing "official" about the "base" user, but the base/archlinux and base/devel images have been there since before all others and are also mentioned in the archlinux wiki page on Docker so they are the most "official" images available.

I do prefer smaller images and there are definitely some ways of cutting down the size of nubs/arch-build, but a lot of the bloat is just due to the size of the packages that are installed using base-devel.

As for finalduty/archlinux.  It looks nice, but is in no way more official.  It's a shame we don't have an official arch linux image in the docker hub.  If we did, things would be so much nicer!

**Comment by bebehei** [comment]
> As for finalduty/archlinux. It looks nice, but is in no way more official. It's a shame we don't have an official arch linux image in the docker hub. If we did, things would be so much nicer!

Yes! docker-library/official-images#2488

> I do prefer smaller images and there are definitely some ways of cutting down the size of nubs/arch-build, but a lot of the bloat is just due to the size of the packages that are installed using base-devel.

Sorry, but I have to decline this. Most of the space is consumed, cause by `pacman -Syu` your image upgrades every package. So every file is **stored twice**. This is self-defeating and an anti-pattern

The worst thing is, that I can understand your general objections. ArchLinux support for docker containers is pretty fucked up currently.

**Comment by bebehei** [comment]
Hey Spencer,

I had been able to get in touch with tmc, who gave me access to `base/devel`. From now on it's getting updated daily and is a mirror of `archimg/base-devel`. So inheriting from `base/devel` is now completely fine and also does not need anymore 5 different pacman calls.

Also I created a git-repo for a docker container with the purpose of building PKGBUILDs. It's available on [archimg/archlinux-pkgbuilder-docker](https://github.com/archimg/archlinux-pkgbuilder-docker).

@archimg is a GitHub organisation and I'd like to build up a team there, which maintains important Archlinux images. If you would like to contribute, I'd highly appreciate it and will give you access there.

Cheers,
Benedikt
